mw-webapp/src/utils/auth/handleUserCredentials.ts

```diff
@@ -1,16 +1,29 @@
 import {getRedirectResult} from "firebase/auth";
-import {writeNewUserData} from "src/utils/auth/writeNewUserData";
 import {auth} from "src/firebase";
+import {UserService} from "src/service/UserService";
+
+const newUserBlankWays = {
+  ownWays: [""],
+  favoriteWays: [""],
+  mentoringWays: [""],
+};
 
 export const handleUserCredentials = async () => {
   try {
     const userCredentials = await getRedirectResult(auth);
-    if (userCredentials) {
-      writeNewUserData(
-        {userId: userCredentials.user.uid, userEmail: userCredentials.user.email, userName: userCredentials.user.displayName},
-      );
-    } else {
-      return;
+    if(userCredentials) {
+      if (userCredentials.user.email && userCredentials.user.displayName) {
```

Comment:
1. nested if - could we avoid it?
2. handleUserCredentials - could we find a little bit more appropriate name, I dont understand what this function should do
3. I dont understand also when I should call this function

Reply:
Rename and delete nested if